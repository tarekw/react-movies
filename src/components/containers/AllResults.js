import AllResults from '../ui/AllResults'
import { connect } from 'react-redux'

const mapStateToProps = (state, props) => ({
    results: state.results,
    handleClick: props.handleClick
});

export default connect(mapStateToProps)(AllResults);
