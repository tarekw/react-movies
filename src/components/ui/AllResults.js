import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';

import GridItem from './GridItem';

const AllResults = ({ results=[], handleClick=f=>f }) => {
    return (
        <Grid container spacing={10} style={{padding: 24}}>
            { results.map(item => (
                <div key={item.id} onClick={() => handleClick(item.id)}>
                    <Grid item xs={12} sm={6} lg={4} xl={3}>
                        <GridItem movie={item} />
                    </Grid>
                </div>
            )) }
        </Grid>
    );
}

AllResults.propTypes = {
    results: PropTypes.array
}

export default AllResults
