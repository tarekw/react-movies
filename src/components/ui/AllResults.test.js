import React from 'react';
import { shallow } from 'enzyme';

import AllResults from './AllResults';

const mockResults = [
    {
        popularity: 622.695,
        vote_count: 1612,
        video: false,
        poster_path: "\/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
        id: 475557,
        adult: false,
        backdrop_path: "\/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg",
        original_language: "en",
        original_title: "Joker",
        genre_ids: [80, 18, 53],
        title: "Joker",
        vote_average: 8.7,
        overview: "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
        release_date: "2019-10-04"
    }, {
        popularity: 203.784,
        vote_count: 1398,
        video: false,
        poster_path: "\/zfE0R94v1E8cuKAerbskfD3VfUt.jpg",
        id: 474350,
        adult: false,
        backdrop_path: "\/8moTOzunF7p40oR5XhlDvJckOSW.jpg",
        original_language: "en",
        original_title: "It Chapter Two",
        genre_ids: [27],
        title: "It Chapter Two",
        vote_average: 7,
        overview: "27 years after overcoming the malevolent supernatural entity Pennywise, the former members of the Losers' Club, who have grown up and moved away from Derry, are brought back together by a devastating phone call.",
        release_date: "2019-09-06"
    }, {
        popularity: 170.756,
        vote_count: 4314,
        video: false,
        poster_path: "\/lcq8dVxeeOqHvvgcte707K0KVx5.jpg",
        id: 429617,
        adult: false,
        backdrop_path: "\/5myQbDzw3l8K9yofUXRJ4UTVgam.jpg",
        original_language: "en",
        original_title: "Spider-Man: Far from Home",
        genre_ids: [28, 12, 878],
        title: "Spider-Man: Far from Home",
        vote_average: 7.6,
        overview: "Peter Parker and his friends go on a summer trip to Europe. However, they will hardly be able to rest - Peter will have to agree to help Nick Fury uncover the mystery of creatures that cause natural disasters and destruction throughout the continent.",
        release_date: "2019-07-02"
    }
];

it('should render correctly with default props', () => {
    const component = shallow(<AllResults />);
    expect(component).toMatchSnapshot();
});

it('should render correctly with mocked props', () => {
    const component = shallow(<AllResults results={mockResults} />);
    expect(component).toMatchSnapshot();
});
