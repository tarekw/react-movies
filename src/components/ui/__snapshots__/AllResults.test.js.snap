// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with default props 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={10}
  style={
    Object {
      "padding": 24,
    }
  }
/>
`;

exports[`should render correctly with mocked props 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={10}
  style={
    Object {
      "padding": 24,
    }
  }
>
  <div
    key="475557"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      sm={6}
      xl={3}
      xs={12}
    >
      <GridItem
        movie={
          Object {
            "adult": false,
            "backdrop_path": "/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg",
            "genre_ids": Array [
              80,
              18,
              53,
            ],
            "id": 475557,
            "original_language": "en",
            "original_title": "Joker",
            "overview": "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
            "popularity": 622.695,
            "poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
            "release_date": "2019-10-04",
            "title": "Joker",
            "video": false,
            "vote_average": 8.7,
            "vote_count": 1612,
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div
    key="474350"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      sm={6}
      xl={3}
      xs={12}
    >
      <GridItem
        movie={
          Object {
            "adult": false,
            "backdrop_path": "/8moTOzunF7p40oR5XhlDvJckOSW.jpg",
            "genre_ids": Array [
              27,
            ],
            "id": 474350,
            "original_language": "en",
            "original_title": "It Chapter Two",
            "overview": "27 years after overcoming the malevolent supernatural entity Pennywise, the former members of the Losers' Club, who have grown up and moved away from Derry, are brought back together by a devastating phone call.",
            "popularity": 203.784,
            "poster_path": "/zfE0R94v1E8cuKAerbskfD3VfUt.jpg",
            "release_date": "2019-09-06",
            "title": "It Chapter Two",
            "video": false,
            "vote_average": 7,
            "vote_count": 1398,
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div
    key="429617"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      sm={6}
      xl={3}
      xs={12}
    >
      <GridItem
        movie={
          Object {
            "adult": false,
            "backdrop_path": "/5myQbDzw3l8K9yofUXRJ4UTVgam.jpg",
            "genre_ids": Array [
              28,
              12,
              878,
            ],
            "id": 429617,
            "original_language": "en",
            "original_title": "Spider-Man: Far from Home",
            "overview": "Peter Parker and his friends go on a summer trip to Europe. However, they will hardly be able to rest - Peter will have to agree to help Nick Fury uncover the mystery of creatures that cause natural disasters and destruction throughout the continent.",
            "popularity": 170.756,
            "poster_path": "/lcq8dVxeeOqHvvgcte707K0KVx5.jpg",
            "release_date": "2019-07-02",
            "title": "Spider-Man: Far from Home",
            "video": false,
            "vote_average": 7.6,
            "vote_count": 4314,
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </div>
</WithStyles(ForwardRef(Grid))>
`;
